src/services/dumpers/agent-nodejs.ts

```diff
@@ -0,0 +1,190 @@
+import AbstractDumper from './abstract-dumper';
+
+export default class AgentNodeJs extends AbstractDumper {
+  private env: { FOREST_SERVER_URL: string; FOREST_URL_IS_DEFAULT: string };
+
+  private readonly DEFAULT_PORT = 3310;
+
+  private readonly isLinuxOs: boolean;
+
+  private readonly isDatabaseLocal: (dbConfig: DbConfigInterface) => boolean;
+
+  private readonly buildDatabaseUrl: (dbConfig: DbConfigInterface) => string;
+
+  private readonly snakeCase: (string: string) => string;
+
+  constructor(context) {
+    const { assertPresent, env, isLinuxOs, buildDatabaseUrl, isDatabaseLocal, snakeCase } = context;
+
+    assertPresent({
+      env,
+      isLinuxOs,
+      isDatabaseLocal,
+      snakeCase,
+    });
+
+    super(context);
+
+    this.env = env;
+    this.isLinuxOs = isLinuxOs;
+    this.buildDatabaseUrl = buildDatabaseUrl;
+    this.isDatabaseLocal = isDatabaseLocal;
+    this.snakeCase = snakeCase;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  protected get templateFolder() {
+    return 'agent-nodejs';
+  }
+
+  writePackageJson(dbDialect: string, applicationName: string) {
+    const dependencies: { [name: string]: string } = {
+      dotenv: '^16.0.1',
+      '@forestadmin/agent': '^1.0.0',
+    };
+
+    if (dbDialect === 'mongodb') {
+      dependencies['@forestadmin/datasource-mongoose'] = '^1.0.0';
+    } else {
+      dependencies['@forestadmin/datasource-sql'] = '^1.0.0';
+    }
+
+    if (dbDialect) {
+      if (dbDialect.includes('postgres')) {
+        dependencies.pg = '~8.2.2';
+        dependencies['pg-hstore'] = '~2.3.4';
+      } else if (dbDialect === 'mysql') {
+        dependencies.mysql2 = '~2.2.5';
+      } else if (dbDialect === 'mariadb') {
+        dependencies.mariadb = '^2.3.3';
+      } else if (dbDialect === 'mssql') {
+        dependencies.tedious = '^6.4.0';
+      }
+    }
+
+    const pkg = {
+      name: toValidPackageName(applicationName),
+      version: '0.0.1',
+      private: true,
+      main: 'index.js',
+      scripts: {
+        'start:watch': 'nodemon',
+        start: 'node ./index.js',
+      },
+      nodemonConfig: {
+        ignore: ['./forestadmin-schema.json'],
+      },
+      dependencies,
+      devDependencies: {
+        nodemon: '^2.0.12',
+      },
+    };
+
+    this.writeFile('package.json', `${JSON.stringify(pkg, null, 2)}\n`);
+  }
+
+  writeIndex(dbDialect: string) {
+    const isMongoose = dbDialect === 'mongodb';
+
+    const context = {
+      isMongoose,
+      isMySQL: dbDialect === 'mysql',
+      isMSSQL: dbDialect === 'mssql',
+      isMariaDB: dbDialect === 'mariadb',
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      datasourceCreation: isMongoose
+        ? 'createMongooseDataSource(connection, {})'
+        : `
+    createSqlDataSource({
+      uri: process.env.DATABASE_URL,
+      schema: process.env.DATABASE_SCHEMA || 'public',
+      ...dialectOptions,
+    }),
```

Comment:
[SKIP] I don't really like this syntax, hurts the eyes a little bit. An if statement would have been less cool but easier to read. I've changed this in a further PR

Reply:
[SKIP] 👍